Src/FluentAssertions/CallerIdentification/AwaitParsingStrategy.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Text;
+
+namespace FluentAssertions.CallerIdentification
+{
+    internal class AwaitParsingStrategy : IParsingStrategy
+    {
+        private const string KeywordToSkip = "await";
+
+        public ParsingState Parse(char symbol, StringBuilder statement)
+        {
+            if (statement.Length >= KeywordToSkip.Length)
+            {
+                var leftIndex = statement.Length - 1;
+                var rightIndex = KeywordToSkip.Length - 1;
+
+                for (var offset = 0; offset < KeywordToSkip.Length; offset++)
+                {
+                    if (statement[leftIndex - offset] != KeywordToSkip[rightIndex - offset])
+                    {
+                        return ParsingState.InProgress;
+                    }
+                }
```

Comment:
I had to read this a couple of times to realise that it's a `Stringbuilder` variant of `EndsWith`.

Reply:
Yeah, copied it from the `ShouldCallParsingStrategy` and then tried to see if I could use spans or the new ranged indexes. But I don't think it helps a lot.